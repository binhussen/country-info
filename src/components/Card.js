const card = () => {
  return (
    <div className="m-1 h-25vh w-47vw p-6 cursor-pointer rounded bg-slate-500">
      <h3 className="text-xl font-medium mx-1">Ethiopia</h3>
      <p>Region : Africa</p>
      <p>Population: 114,0000</p>
    </div>
  )
}

export default card;