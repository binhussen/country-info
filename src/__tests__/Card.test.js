/** @format */

import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Card from '../components/Card';

const country = {
  name: {
    common: 'Ethiopia',
    official: 'Federal Democratic Republic of Ethiopia',
    nativeName: {
      amh: {
        official: 'የኢትዮጵያ ፌዴራላዊ ዲሞክራሲያዊ ሪፐብሊክ',
        common: 'ኢትዮጵያ',
      },
    },
  },
  code: 'ET',
  currency: {
    ETB: {
      name: 'Ethiopian birr',
      symbol: 'Br',
    },
  },
  capital: ['Addis Ababa'],
  region: 'Africa',
  subregion: 'Eastern Africa',
  languages: {
    amh: 'Amharic',
  },
  borders: ['DJI', 'ERI', 'KEN', 'SOM', 'SSD', 'SDN'],
  area: 1104300,
  map: {
    googleMaps: 'https://goo.gl/maps/2Q4hQWCbhuZLj3fG6',
    openStreetMaps: 'https://www.openstreetmap.org/relation/192800',
  },
  population: 114963583,
  timezones: ['UTC+03:00'],
  flag: {
    png: 'https://flagcdn.com/w320/et.png',
    svg: 'https://flagcdn.com/et.svg',
  },
};

describe('Render Card', () => {
  test('card snapshot', () => {
    const card = render(
      <BrowserRouter>
        <Card country={country} />
      </BrowserRouter>,
    );
    expect(card).toMatchSnapshot();
  });
});
