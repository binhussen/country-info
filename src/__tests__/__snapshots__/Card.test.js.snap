// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Card card snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="text-white transition-shadow opacity-100 bg bg-slate-400 shadow-2xl"
        href="/weather/Ethiopia"
      >
        <div
          class="flex items-end flex-col justify-between p-4 shadow-2xl h-25vh"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.0519 14.8285L13.4661 16.2427L17.7088 12L13.4661 7.7574L12.0519 9.17161L13.8804 11H6.34321V13H13.8803L12.0519 14.8285Z"
              fill="currentColor"
            />
            <path
              clip-rule="evenodd"
              d="M19.7782 19.7782C24.0739 15.4824 24.0739 8.51759 19.7782 4.22183C15.4824 -0.0739417 8.51759 -0.0739417 4.22183 4.22183C-0.0739417 8.51759 -0.0739417 15.4824 4.22183 19.7782C8.51759 24.0739 15.4824 24.0739 19.7782 19.7782ZM18.364 18.364C21.8787 14.8492 21.8787 9.15076 18.364 5.63604C14.8492 2.12132 9.15076 2.12132 5.63604 5.63604C2.12132 9.15076 2.12132 14.8492 5.63604 18.364C9.15076 21.8787 14.8492 21.8787 18.364 18.364Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="text-right uppercase"
          >
            <h3
              class="text-xl font-medium mx-1"
            >
              Ethiopia
               
              ኢትዮጵያ
            </h3>
            <p>
              1104300
            </p>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="text-white transition-shadow opacity-100 bg bg-slate-400 shadow-2xl"
      href="/weather/Ethiopia"
    >
      <div
        class="flex items-end flex-col justify-between p-4 shadow-2xl h-25vh"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.0519 14.8285L13.4661 16.2427L17.7088 12L13.4661 7.7574L12.0519 9.17161L13.8804 11H6.34321V13H13.8803L12.0519 14.8285Z"
            fill="currentColor"
          />
          <path
            clip-rule="evenodd"
            d="M19.7782 19.7782C24.0739 15.4824 24.0739 8.51759 19.7782 4.22183C15.4824 -0.0739417 8.51759 -0.0739417 4.22183 4.22183C-0.0739417 8.51759 -0.0739417 15.4824 4.22183 19.7782C8.51759 24.0739 15.4824 24.0739 19.7782 19.7782ZM18.364 18.364C21.8787 14.8492 21.8787 9.15076 18.364 5.63604C14.8492 2.12132 9.15076 2.12132 5.63604 5.63604C2.12132 9.15076 2.12132 14.8492 5.63604 18.364C9.15076 21.8787 14.8492 21.8787 18.364 18.364Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="text-right uppercase"
        >
          <h3
            class="text-xl font-medium mx-1"
          >
            Ethiopia
             
            ኢትዮጵያ
          </h3>
          <p>
            1104300
          </p>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Render Card matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="text-white transition-shadow opacity-100 bg bg-slate-400 shadow-2xl"
        href="/weather/Ethiopia"
      >
        <div
          class="flex items-end flex-col justify-between p-4 shadow-2xl h-25vh"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.0519 14.8285L13.4661 16.2427L17.7088 12L13.4661 7.7574L12.0519 9.17161L13.8804 11H6.34321V13H13.8803L12.0519 14.8285Z"
              fill="currentColor"
            />
            <path
              clip-rule="evenodd"
              d="M19.7782 19.7782C24.0739 15.4824 24.0739 8.51759 19.7782 4.22183C15.4824 -0.0739417 8.51759 -0.0739417 4.22183 4.22183C-0.0739417 8.51759 -0.0739417 15.4824 4.22183 19.7782C8.51759 24.0739 15.4824 24.0739 19.7782 19.7782ZM18.364 18.364C21.8787 14.8492 21.8787 9.15076 18.364 5.63604C14.8492 2.12132 9.15076 2.12132 5.63604 5.63604C2.12132 9.15076 2.12132 14.8492 5.63604 18.364C9.15076 21.8787 14.8492 21.8787 18.364 18.364Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="text-right uppercase"
          >
            <h3
              class="text-xl font-medium mx-1"
            >
              Ethiopia
               
              ኢትዮጵያ
            </h3>
            <p>
              1104300
            </p>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="text-white transition-shadow opacity-100 bg bg-slate-400 shadow-2xl"
      href="/weather/Ethiopia"
    >
      <div
        class="flex items-end flex-col justify-between p-4 shadow-2xl h-25vh"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.0519 14.8285L13.4661 16.2427L17.7088 12L13.4661 7.7574L12.0519 9.17161L13.8804 11H6.34321V13H13.8803L12.0519 14.8285Z"
            fill="currentColor"
          />
          <path
            clip-rule="evenodd"
            d="M19.7782 19.7782C24.0739 15.4824 24.0739 8.51759 19.7782 4.22183C15.4824 -0.0739417 8.51759 -0.0739417 4.22183 4.22183C-0.0739417 8.51759 -0.0739417 15.4824 4.22183 19.7782C8.51759 24.0739 15.4824 24.0739 19.7782 19.7782ZM18.364 18.364C21.8787 14.8492 21.8787 9.15076 18.364 5.63604C14.8492 2.12132 9.15076 2.12132 5.63604 5.63604C2.12132 9.15076 2.12132 14.8492 5.63604 18.364C9.15076 21.8787 14.8492 21.8787 18.364 18.364Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="text-right uppercase"
        >
          <h3
            class="text-xl font-medium mx-1"
          >
            Ethiopia
             
            ኢትዮጵያ
          </h3>
          <p>
            1104300
          </p>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Render Card render card 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="text-white transition-shadow opacity-100 bg bg-slate-400 shadow-2xl"
        href="/weather/Ethiopia"
      >
        <div
          class="flex items-end flex-col justify-between p-4 shadow-2xl h-25vh"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.0519 14.8285L13.4661 16.2427L17.7088 12L13.4661 7.7574L12.0519 9.17161L13.8804 11H6.34321V13H13.8803L12.0519 14.8285Z"
              fill="currentColor"
            />
            <path
              clip-rule="evenodd"
              d="M19.7782 19.7782C24.0739 15.4824 24.0739 8.51759 19.7782 4.22183C15.4824 -0.0739417 8.51759 -0.0739417 4.22183 4.22183C-0.0739417 8.51759 -0.0739417 15.4824 4.22183 19.7782C8.51759 24.0739 15.4824 24.0739 19.7782 19.7782ZM18.364 18.364C21.8787 14.8492 21.8787 9.15076 18.364 5.63604C14.8492 2.12132 9.15076 2.12132 5.63604 5.63604C2.12132 9.15076 2.12132 14.8492 5.63604 18.364C9.15076 21.8787 14.8492 21.8787 18.364 18.364Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="text-right uppercase"
          >
            <h3
              class="text-xl font-medium mx-1"
            >
              Ethiopia
               
              ኢትዮጵያ
            </h3>
            <p>
              1104300
            </p>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="text-white transition-shadow opacity-100 bg bg-slate-400 shadow-2xl"
      href="/weather/Ethiopia"
    >
      <div
        class="flex items-end flex-col justify-between p-4 shadow-2xl h-25vh"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.0519 14.8285L13.4661 16.2427L17.7088 12L13.4661 7.7574L12.0519 9.17161L13.8804 11H6.34321V13H13.8803L12.0519 14.8285Z"
            fill="currentColor"
          />
          <path
            clip-rule="evenodd"
            d="M19.7782 19.7782C24.0739 15.4824 24.0739 8.51759 19.7782 4.22183C15.4824 -0.0739417 8.51759 -0.0739417 4.22183 4.22183C-0.0739417 8.51759 -0.0739417 15.4824 4.22183 19.7782C8.51759 24.0739 15.4824 24.0739 19.7782 19.7782ZM18.364 18.364C21.8787 14.8492 21.8787 9.15076 18.364 5.63604C14.8492 2.12132 9.15076 2.12132 5.63604 5.63604C2.12132 9.15076 2.12132 14.8492 5.63604 18.364C9.15076 21.8787 14.8492 21.8787 18.364 18.364Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="text-right uppercase"
        >
          <h3
            class="text-xl font-medium mx-1"
          >
            Ethiopia
             
            ኢትዮጵያ
          </h3>
          <p>
            1104300
          </p>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
